libs/auth/token_loader.go

```diff
@@ -0,0 +1,83 @@
+package auth
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/databricks/databricks-sdk-go/credentials/u2m"
+	"github.com/databricks/databricks-sdk-go/credentials/u2m/cache"
+	"golang.org/x/oauth2"
+)
+
+type persistentAuth interface {
+	Token() (*oauth2.Token, error)
+	Close() error
+}
+
+var persistentAuthFactory = func(ctx context.Context, opts ...u2m.PersistentAuthOption) (persistentAuth, error) {
+	return u2m.NewPersistentAuth(ctx, opts...)
+}
```

Comment:
If I understand correctly, this is defined purely for a single unit test that checks if the `PersistentAuthOpts` aren't mutated, correct?

I'd get rid of this as we don't need to do that. Please check my next comment for more details 👍